backend/s3/file.go

```diff
@@ -534,8 +533,42 @@ func waitUntilFileExists(file vfs.File, retries int) error {
 	return nil
 }
 
-var getDownloader = func(client s3iface.S3API, partSize int64) Downloader {
-	return s3manager.NewDownloaderWithClient(client, func(d *s3manager.Downloader) {
-		d.PartSize = partSize
-	})
+func (f *File) getReader() (io.ReadCloser, error) {
+	if f.reader == nil {
+		// Create the request to get the object
+		input := new(s3.GetObjectInput).
+			SetBucket(f.bucket).
+			SetKey(f.key).
+			SetRange(fmt.Sprintf("bytes=%d-", f.cursorPos))
```

Comment:
This is the key change.  We use the cursor position as the startpoint for reading from getObject stream.  Anytime Seek or Close called, we invalidate the reader and any new call to Read opens a new stream at the updated cursor position.